File: VB/NavigateUri.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows
Imports DevExpress.Xpf.LayoutControl

Namespace DXSilverlightApplication1
	Public Class NavigateUri
		Public Shared ReadOnly NavigateUriProperty As DependencyProperty = DependencyProperty.RegisterAttached("NavigateUri", GetType(Uri), GetType(NavigateUri), New PropertyMetadata(Nothing, New PropertyChangedCallback(AddressOf OnNavigateUriChanged)))

		Public Shared Function GetNavigateUri(ByVal target As Tile) As Uri
			Return CType(target.GetValue(NavigateUriProperty), Uri)
		End Function

		Public Shared Sub SetNavigateUri(ByVal target As Tile, ByVal value As Uri)
			target.SetValue(NavigateUriProperty, value)
		End Sub

		Private Shared Sub OnNavigateUriChanged(ByVal o As DependencyObject, ByVal e As DependencyPropertyChangedEventArgs)
			Dim tile As Tile = CType(o, Tile)
			RemoveHandler tile.Click, AddressOf o_Click
			AddHandler tile.Click, AddressOf o_Click
		End Sub
		Private Shared Sub o_Click(ByVal sender As Object, ByVal e As EventArgs)
			System.Windows.Browser.HtmlPage.Window.Navigate(CType((TryCast(sender, DependencyObject)).GetValue(NavigateUriProperty), Uri))
		End Sub
	End Class
End Namespace

File: CS/NavigateUri.cs
using System;
using System.Windows;
using DevExpress.Xpf.LayoutControl;

namespace DXSilverlightApplication1
{
    public class NavigateUri
    {
        public static readonly DependencyProperty NavigateUriProperty = DependencyProperty.RegisterAttached("NavigateUri", typeof(Uri), typeof(NavigateUri), new PropertyMetadata(null, new PropertyChangedCallback(OnNavigateUriChanged)));

        public static Uri GetNavigateUri(Tile target)
        {
            return (Uri)target.GetValue(NavigateUriProperty);
        }

        public static void SetNavigateUri(Tile target, Uri value)
        {
            target.SetValue(NavigateUriProperty, value);
        }

        private static void OnNavigateUriChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
        {
            Tile tile = (Tile)o;
            tile.Click -= o_Click;
            tile.Click += o_Click;
        }
        private static void o_Click(object sender, EventArgs e)
        {
            System.Windows.Browser.HtmlPage.Window.Navigate((Uri)(sender as DependencyObject).GetValue(NavigateUriProperty));
        }    
    }
}
